src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+/* eslint-disable @typescript-eslint/lines-between-class-members */
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle';
+  color: 'red' | 'green' | 'blue';
+  private a: number;
+  private b: number;
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    const sides = [a, b, c].sort((x, z) => z - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error('This triangle cannot exist');
+    }
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Rectangle implements Figure {}
+export class Circle implements Figure {
+  shape = 'circle';
+  color: 'red' | 'green' | 'blue';
+  private radius: number;
+
+  constructor(color: 'red' | 'green' | 'blue', radius: number) {
+    this.color = color;
+
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+    this.radius = radius;
+  }
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape = 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  private width: number;
+  private height: number;
+
+  constructor(color: 'red' | 'green' | 'blue', width: number, height: number) {
```

Comment:
for better readability
```suggestion
  constructor(
    color: Color,
    width: number, 
    height: number
  ) {
```